ledfx/utils.py

```diff
@@ -1168,3 +1168,45 @@ def wled_support_DDP(build) -> bool:
         return True
     else:
         return False
+
+
+name_to_icon = {
+    "Desk": "mdi:desk",
+    "Desktop": "mdi:desktop-classic",
+    "Monitor": "mdi:monitor",
+    "TV": "mdi:television",
+    "Kitchen": "mdi:stove",
+    "Fridge": "mdi:fridge",
+    "Bed": "mdi:bed",
+    "Radiator": "mdi:radiator",
+    "Heater": "mdi:radiator",
+    "Cinema": "mdi:theater",
+    "Water": "mdi:water-pump",
+    "Matrix": "mdi:table-large",
+    "Wardrobe": "mdi:wardrobe",
+    "Cupboard": "mdi:cupboard",
+    "Speaker": "mdi:speaker",
+    "Chair": "mdi:chair-rolling",
+    "Couch": "mdi:sofa",
+    "Sofa": "mdi:sofa",
+    "Cloud": "mdi:cloud",
+    "Roof": "mdi:home-roof",
+    "Logo-III-Y": "yz:logo3y",
+    "Logo-III-Z": "yz:logo3z",
+    "Logo-III-Top": "yz:logo3top",
+    "Logo-III-Left": "yz:logo3left",
+    "Logo-III-Right": "yz:logo3right",
+    "Logo-III": "yz:logo3",
+    "Logo-II-Y": "yz:logo2y",
+    "Logo-II-Z": "yz:logo2z",
+    "Logo-II-Top": "yz:logo2top",
+    "Logo-II-Bottom": "yz:logo2bot",
+    "Logo-II": "yz:logo2",
+}
+
+
+def get_icon_name(wled_name):
+    for name, icon in name_to_icon.items():
```

Comment:
As name_to_icon is a dict, you can just look up the key, you don't need to search through with a loop.

return name_to_icon.get(wled_name, "wled")

This "should" find the matching key for wled_name and return the icon, or otherwise default to "wled"

At this point I would just inline that code where it is called, as making it a function is just obscuration





Secondary concern is where this list of icons lives / managed, feels like front end leaking into back end

Reply:
If your change works, that would be SUPER NICE, testing asap.
some questions bro @bigredfrog 
1) Would it check for string containing? So if a user names his wled `Main-Desk` will it still find the icon with the key `Desk` ?
2) about the backend / frontend part.
its like the ones we did for launchpad:
https://github.com/LedFx/LedFx/blob/main/ledfx/devices/launchpad_lib.py#L689
https://github.com/LedFx/LedFx/blob/main/ledfx/devices/launchpad_lib.py#L692
https://github.com/LedFx/LedFx/blob/main/ledfx/devices/launchpad_lib.py#L707
its only used for scanned WLEDs, and just sets the initial icon.
I would gladly maintain this within the frontend, but then more python voodoo was needed.
I can easily provide the object inside the body of the post call. Can u help me, to use that?

Reply:
But when served from frontend, the backend would not have it inside the scan_on_startup:
https://github.com/LedFx/LedFx/blob/main/ledfx/core.py#L269

Reply:
oh you are looking for a best match, not an actual match? That is indeed fluffier, there may be a pythonic way to do that, but if you brute force works, all good.

I can easily provide the object inside the body of the post call. Can u help me, to use that? 

I will just need example and understanding of what you are after and we will make it work...

Reply:
Before you worry about prototyping "something else" let me have another look at this, this evening